tests/components/nws/test_init.py

```diff
@@ -0,0 +1,67 @@
+"""Tests for init module."""
+from homeassistant.components import nws
+from homeassistant.components.nws.const import CONF_STATION, DOMAIN
+from homeassistant.const import CONF_API_KEY, CONF_LATITUDE, CONF_LONGITUDE
+from homeassistant.setup import async_setup_component
+
+from tests.common import assert_setup_component
+from tests.components.nws.const import MINIMAL_CONFIG
+
+LATLON_CONFIG = {
+    DOMAIN: [{CONF_API_KEY: "test", CONF_LATITUDE: 45.0, CONF_LONGITUDE: -75.0}]
+}
+FULL_CONFIG = {
+    DOMAIN: [
+        {
+            CONF_API_KEY: "test",
+            CONF_LATITUDE: 45.0,
+            CONF_LONGITUDE: -75.0,
+            CONF_STATION: "XYZ",
+        }
+    ]
+}
+DUPLICATE_CONFIG = {
+    DOMAIN: [
+        {CONF_API_KEY: "test", CONF_LATITUDE: 45.0, CONF_LONGITUDE: -75.0},
+        {CONF_API_KEY: "test", CONF_LATITUDE: 45.0, CONF_LONGITUDE: -75.0},
+    ]
+}
+
+
+async def test_no_config(hass, mock_simple_nws):
+    """Test that nws does not setup with no config."""
+    with assert_setup_component(0):
+        assert await async_setup_component(hass, DOMAIN, {}) is True
+        assert DOMAIN not in hass.data
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5329046)